src/partition/ReceivedPartition.cpp

```diff
@@ -470,143 +488,312 @@ void ReceivedPartition::createOwnerInformation()
   PRECICE_TRACE();
   Event e("partition.createOwnerInformation." + _mesh->getName(), precice::syncMode);
 
-  if (utils::MasterSlave::isSlave()) {
+  if (m2n().usesTwoLevelInitialization()) {
+    /*
+    1- receive local bb map from master
+    2- filter bb map to keep the connected ranks
+    3- own the vertices that only fits into the ranks bb
+    4- send number of owned vertices and shared vertices list to neighbors
+    5- for the remaining vertices: check if we have less vertexes -> own it!
+    */
```

Comment:
Could you please additionally add a few lines of big-picture explanation here. What is done and why.
The steps you have are already very helpful, but still require the reader to understand how you got there.

Reply:
Explanation is added. Please check for correctness and completeness.